tests/vmi_networking_test.go

```diff
@@ -721,43 +721,50 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
-				Expect(err).ToNot(HaveOccurred(), "Failed to traceroute to VMI %s within the given timeout", serverVMI.Name)
-			} else {
-				if ipv4NetworkCIDR == "" {
-					ipv4NetworkCIDR = api.DefaultVMCIDR
+			Expect(len(serverVMI.Status.Interfaces[0].IPs)).Should(BeNumerically(">", 0))
+
+			for _, serverIP := range serverVMI.Status.Interfaces[0].IPs {
+				if netutils.IsIPv6String(serverIP) {
+					if netutils.IsIPv6String(serverIP) {
+						Skip("Skip till dual stack feature ready (PR on hold)")
+					}
+
+					By("Testing IPv6")
+					By("Checking traceroute (IPv6) from vmi to cluster nodes gateway")
+					// Cluster nodes subnet (docker network gateway)
+					// Docker network subnet cidr definition https://bit.ly/2wZTgMK
+					err := tests.CheckForTextExpecter(serverVMI, createExpectTraceroute6("2001:db8:1::1"), 30)
+					Expect(err).ToNot(HaveOccurred(), "Failed to traceroute to VMI %s within the given timeout", serverVMI.Name)
+				} else {
+					By("Testing IPv4")
+					if ipv4NetworkCIDR == "" {
+						ipv4NetworkCIDR = api.DefaultVMCIDR
+					}
+					By("Checking ping (IPv4) to gateway")
+					ipAddr := gatewayIPFromCIDR(ipv4NetworkCIDR)
+					Expect(tests.PingFromVMConsole(serverVMI, ipAddr)).To(Succeed())
+
+					By("Checking ping (IPv4) to google")
+					Expect(tests.PingFromVMConsole(serverVMI, "8.8.8.8")).To(Succeed())
+					Expect(tests.PingFromVMConsole(clientVMI, "google.com")).To(Succeed())
 				}
-				By("Checking ping to gateway")
-				ipAddr := gatewayIPFromCIDR(ipv4NetworkCIDR)
-				Expect(tests.PingFromVMConsole(serverVMI, ipAddr)).To(Succeed())
-
-				By("Checking ping to google")
-				Expect(tests.PingFromVMConsole(serverVMI, "8.8.8.8")).To(Succeed())
-				Expect(tests.PingFromVMConsole(clientVMI, "google.com")).To(Succeed())
-			}
 
-			By("starting a tcp server")
-			tcpPort := "8080"
-			err = tests.CheckForTextExpecter(serverVMI, createExpectStartTcpServer(tcpPort), 30)
-			Expect(err).ToNot(HaveOccurred())
+				By("starting a tcp server")
+				tcpPort := "8080"
+				err = tests.CheckForTextExpecter(serverVMI, createExpectStartTcpServer(tcpPort), 30)
+				Expect(err).ToNot(HaveOccurred())
 
-			By("Connecting from the client vm")
-			err = tests.CheckForTextExpecter(clientVMI, createExpectConnectToServer(serverIP, tcpPort, true), 30)
-			Expect(err).ToNot(HaveOccurred())
+				By("Connecting from the client vm")
+				err = tests.CheckForTextExpecter(clientVMI, createExpectConnectToServer(serverIP, tcpPort, true), 30)
+				Expect(err).ToNot(HaveOccurred())
 
-			By("Rejecting the connection from the client to unregistered port")
-			err = tests.CheckForTextExpecter(clientVMI, createExpectConnectToServer(serverIP, "8081", false), 30)
-			Expect(err).ToNot(HaveOccurred())
+				By("Rejecting the connection from the client to unregistered port")
+				err = tests.CheckForTextExpecter(clientVMI, createExpectConnectToServer(serverIP, "8081", false), 30)
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
It should fail to connect here right?
maybe format a message here that explains what should have happen in case it fails?  
I am not sure it will be clear enough from the error returning from `CheckForTextExpecter`.
something like:
`Expect(err).ToNot(HaveOccurred(), "should fail to connect to unregistered port")`

wasn't sure about line 759 and line 755, maybe there a message will help as well, but not sure

Reply:
hmm, i just idented those line, can we take it to other PR ?
its not connected to the title and to the PR purpose

Reply:
yes ,you are right. maybe not so related here